### PR TITLE
Use React 16 portal to render map overlay

### LINKED ISSUE #642: Context (new API) is not propagated to google-map-react children
When using the new context API, google-map-react `children` cannot consume the context because `GoogleMapMarkers` is rendered outside of the rest of the component tree. 

A similar issue was created for the previous context API: https://github.com/google-map-react/google-map-react/issues/64 and was resolved by using `unstable_renderSubtreeIntoContainer`.

However, deprecated `unstable_renderSubtreeIntoContainer` does not work for the new context API.  For the new context API to be propagated properly, `ReactDOM.createPortal` from react 16 needs to be used instead.

**Expected Behaviour**
It's expected that when wrapping the Google map component with a context provider, the children to the Google map component will be able to consume the context

**Current Behaviour**
Children of the Google map component cannot consume the context provided from higher up in the tree.

### PR DESCRIPTION
**Pull request by clayne11**
This allows the new context API to propagate context properly.

Fixes #642.

### CODE REVIEW

File: src/google_map.js [hunk 1]
```diff
@@ -239,7 +239,7 @@ export default class GoogleMap extends Component {
     this.zoomAnimationInProgress_ = false;
 
     this.state = {
-      overlayCreated: false,
+      div: null,
```
Comment (itsmichaeldiego) [inline src/google_map.js]: Can we rename `div` to `overlay`?

**Comment by clayne11** [comment]
Done.

**Comment by stuartherring** [comment]
This change breaks the component when used with React versions prior to 16.

**Comment by itsmichaeldiego** [comment]
@stuartherring @abumusamq Sorry for the inconvenience! This was totally my bad, I just fixed this and released in `1.0.8`.

@clayne11 do you mind opening the PR again? We will add this to the next major release.

Or, I think there should be a way to identify if the user has React 16 or not and use createPortal if so, but that would be a bit more tricky.

**Comment by clayne11** [comment]
No problem I can do that. Are you planning to cut a new major release with this soon? React 16 has been out for a year now (released Sept 26, 2017) so I think it's about time to move forward.

**Comment by itsmichaeldiego** [comment]
@clayne11 Yes, we would like to do that soon but lately it has been only me taking care of the library, any help would be accepted lovely xD.

**Pull request merged**